plugin.lua

```diff
@@ -5,54 +5,17 @@
 -- what do the different prefixes for gmatch results mean:
 -- s = start, f = finish, p = position, no prefix = an actual string capture
 
+---Dev Notes: confirm "path/to/lua-language-server/script/", in Lua.Workspace.Library for completions
+
 -- allow for require to search relative to this plugin file
 -- open for improvements!
-
----Notes: confirm "path/to/lua-language-server/script/", in Lua.Workspace.Library
----Notes: confirm "path/to/bee" in Lua.Workspace.Library
-
-if not __plugin_dev and not _G.__factorio_plugin_initialized then
-  _G.__factorio_plugin_initialized = true
-
-  local config = require("config")
-  local fs = require("bee.filesystem")
-  local workspace = require("workspace")
-
-  -- now it's getting incredibly hacky, I should look into making a PR
-  local is_2_6_0_or_later = debug.getinfo(config.get, "u").nparams > 1
-  local plugin_path
-  if is_2_6_0_or_later then
-    local info = debug.getinfo(3, "uf")
-    local scp
-    for i = 1, info.nups do
-      local name, value = debug.getupvalue(info.func, i)
-      if name == "scp" then
-        scp = value
-      end
-    end
-    if not scp then
-      for i = 1, info.nparams do
-        local name, value = debug.getlocal(3, i)
-        if name == "scp" then
-          scp = value
-        end
-      end
-    end
-    assert(scp, "Unable to get currently used scope/folder. This is very most likely \z
-      caused by internal changes of the language server \z
-      in which case the plugin needs to be changed/updated."
-    )
-
-    plugin_path = fs.path(workspace.getAbsolutePath(scp.uri, config.get(scp.uri, 'Lua.runtime.plugin')))
-  else -- sumneko.lua < 2.6.0
-    ---@diagnostic disable-next-line: missing-parameter
-    plugin_path = fs.path(workspace.getAbsolutePath(config.get('Lua.runtime.plugin')))
-  end
-
-  local new_path = (plugin_path:parent_path() / "?.lua"):string() --[[@as string]]
-  if not package.path:find(new_path, 1, true) then
-    package.path = package.path..";"..new_path
-  end
+local fs = require("bee.filesystem")
+local workspace = require("workspace")
+local scope = require("workspace.scope")
+local plugin_path = fs.path(scope.getScope(workspace.rootUri):get('pluginPath'))
+local new_path = (plugin_path:parent_path() / "?.lua"):string()
+if not package.path:find(new_path, 1, true) then
+  package.path = package.path..";"..new_path
```

Comment:
I still don't get scopes in `sumneko.lua`. Like what you have here looks good but since I don't know what the point of scopes are (and you are using a scope here of course) I don't know if I can trust it

Reply:
It can't be trusted. I think scope is only valid if the plugin is part of an opened workspace or library. At this point I should just PR the path in sumneko.